service/history/historyEngine.go

```diff
@@ -2912,6 +2913,43 @@ func (e *historyEngineImpl) ReapplyEvents(
 		})
 }
 
+func (e *historyEngineImpl) RefreshWorkflowTasks(
+	ctx ctx.Context,
+	domainUUID string,
+	execution workflow.WorkflowExecution,
+) (retError error) {
+
+	context, release, err := e.historyCache.getOrCreateWorkflowExecution(ctx, domainUUID, execution)
+	if err != nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	mutableState, err := context.loadWorkflowExecution()
+	if err != nil {
+		return err
+	}
+	mutableStateTaskRefresher := newMutableStateTaskRefresher(
+		e.shard.GetConfig(),
+		e.shard.GetDomainCache(),
+		e.shard.GetEventsCache(),
+		e.shard.GetLogger(),
+	)
+
+	now := e.shard.GetTimeSource().Now()
+
+	err = mutableStateTaskRefresher.refreshTasks(now, mutableState)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = mutableState.CloseTransactionAsMutation(now, transactionPolicyPassive)
```

Comment:
Why discard the results? `CloseTransactionAsMutation` will return a `WorkflowMutation` object which needs to be persisted. Also I think if passive policy is used, timer task for user timer and activity won't be generated.

Reply:
fixed